go-controller/pkg/node/gateway.go

```diff
@@ -238,9 +241,7 @@ func (g *gateway) AddEgressIP(eip *egressipv1.EgressIP) error {
 		return err
 	}
 	if isSyncRequired {
-		if err = g.Reconcile(); err != nil {
```

Comment:
not sure I understand this change. If this fails, it would retry as part of the retry framework wouldn't it? You are moving it to a separate goroutine and operating outside of retry framework now?

Reply:
I didn't give this much thought as I took it in from the latest rebase of the EgressIP changes.

I had the chance to operate asynchoronously, pretty much as what `g.openflowManager.requestFlowSync()` is doing below, so I took it. `AddEgressIP` is also making use of other asynchronous downstream components, like address manager.

Please note that this change here did not motivate making `Reconcile` asynchronous, we call `Reconcile` from other places where we had no retry mechanism in place.